libs/@local/graph/authorization/src/policies/resource/entity.rs

```diff
@@ -22,7 +22,8 @@ use crate::policies::cedar::{
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 pub struct EntityResource<'a> {
     pub id: EntityId,
-    pub entity_type: Cow<'a, [VersionedUrl]>,
+    pub entity_types: Cow<'a, [VersionedUrl]>,
+    pub entity_base_types: Cow<'a, [BaseUrl]>,
```

Comment:
just for clarification - the base types are the types without any version, correct? In that case, what happens if an entity type happens to inherit from 2 different versions of the same entity type? Will the `base_types` be deduplicated? Or will it always be `assert_eq!(entity_types.len(), entity_base_types.len())`.

Reply:
I'm not sure if we would even allow this scenario, but conceptually, this is converted to a `Set` in Cedar, so it would be deduplicated. The general goal is to have "Is this entity some kind of `User`", we mainly use a `contains` operation. Cedar also does not support closures, this is why we cannot use something like `.find(|type| type == base_url)` but need a dedicated list.